tests/test_tenant_unauthorized.py

```diff
@@ -0,0 +1,37 @@
+from unittest.mock import MagicMock
+import pytest
+from assertpy import assert_that
+from dataall.base.api import bootstrap
+from dataall.base.loader import load_modules, ImportMode
+from dataall.base.context import RequestContext, get_context
+from threading import local
+from dataall.base.db.exceptions import TenantUnauthorized
+
+from pytest import fixture
+
+load_modules(modes={ImportMode.API})
+
+CHECK_PERMS = ['archiveOrganization']
+
+
+@fixture(scope='function')
+def patch_context(mocker, db, userNoTenantPermissions, groupNoTenantPermissions):
+    mock = local()
+    mock.context = RequestContext(
+        db, userNoTenantPermissions.username, [groupNoTenantPermissions.groupUri], userNoTenantPermissions
+    )
+    mocker.patch('dataall.base.context._request_storage', mock)
+
+
+@pytest.mark.parametrize(
+    'name,field_resolver',
+    [
+        (f'{_type.name}.{field.name}', field.resolver)
+        for _type in bootstrap().types
+        for field in _type.fields
+        if field.resolver and _type.name in ['Mutation', 'Query'] and field.name in CHECK_PERMS
```

Comment:
I think we should inverse the for-loop and make pytest fail in case any of the listed CHECK_PERMS queries are missing. i.e...

```python
ALL_RESOLVERS = {
    f'{_type.name}.{field.name}': field.resolver
    for _type in bootstrap().types
    for field in _type.fields
    if field.resolver
}

CHECK_PERMS = [
    'Mutation.archiveOrganization',
    'Mutation.createOrganization',
]


@pytest.mark.parametrize('field_name,resolver', [(perm, ALL_RESOLVERS[perm]) for perm in CHECK_PERMS])
```

Reply:
I implemented something similar in https://github.com/data-dot-all/dataall/pull/1694/commits/75d30c148250894bf791db2bef905c347ed8a3b0 adding an assertion to check that the resolver does not exist but continue for other tests